src/main/java/uk/gov/hmcts/reform/blobrouter/tasks/processors/ContainerProcessor.java

```diff
@@ -76,68 +74,26 @@ private boolean isReady(BlobItem blobClient, String containerName) {
     }
 
     private void processBlob(BlobClient blobClient) {
-        getLastEnvelope(blobClient)
+        findEnvelopeNotInCreatedStatus(blobClient)
             .ifPresentOrElse(
-                envelope -> {
-                    if (envelope.status == Status.CREATED) {
-                        leaseAndThen(blobClient, () ->
-                            continueProcessingEnvelopeIfEligible(
-                                blobClient,
-                                this::logEnvelopeDeleted
-                            )
-                        );
-                    } else {
-                        logEnvelopeAlreadyProcessed(envelope);
-                    }
-                },
-                () -> leaseAndThen(blobClient, () ->
-                    continueProcessingEnvelopeIfEligible(
-                        blobClient,
-                        blobProcessor::process
-                    )
-                )
+                this::logEnvelopeAlreadyProcessed,
+                () -> leaseAndProcess(blobClient)
             );
     }
 
-    private void continueProcessingEnvelopeIfEligible(
-        BlobClient blobClient,
-        Consumer<BlobClient> nonExistingEnvelopeHandler
-    ) {
-        getLastEnvelope(blobClient)
-            .ifPresentOrElse(
-                envelope -> continueProcessingIfPossible(blobClient, envelope),
-                () -> nonExistingEnvelopeHandler.accept(blobClient)
-            );
-    }
-
-    private void continueProcessingIfPossible(BlobClient blobClient, Envelope envelope) {
-        if (envelope.status == Status.CREATED) {
-            blobProcessor.continueProcessing(envelope.id, blobClient);
-        } else {
-            logEnvelopeAlreadyProcessed(envelope);
-        }
-    }
-
-    private void logEnvelopeDeleted(BlobClient blobClient) {
-        logger.error(
-            "Envelope deleted in system for blob {}, container {}",
-            blobClient.getBlobName(), blobClient.getContainerName()
-        );
-    }
-
     private void logEnvelopeAlreadyProcessed(Envelope envelope) {
-        logger.info("Envelope already processed in system, skipping. {} ", envelope.getBasicInfo());
+        logger.info("Envelope already processed in db, skipping. {} ", envelope.getBasicInfo());
     }
 
-    private Optional<Envelope> getLastEnvelope(BlobClient blobClient) {
+    private Optional<Envelope> findEnvelopeNotInCreatedStatus(BlobClient blobClient) {
         return envelopeService
-            .findLastEnvelope(blobClient.getBlobName(), blobClient.getContainerName());
+            .findEnvelopeNotInCreatedStatus(blobClient.getBlobName(), blobClient.getContainerName());
     }
 
-    private void leaseAndThen(BlobClient blobClient, Runnable action) {
+    private void leaseAndProcess(BlobClient blobClient) {
         leaseAcquirer.ifAcquiredOrElse(
```

Comment:
After acquiring lease we need to ensure again that there is no envelope NOT in CREATE state - to eliminate any consequences of race condition.

Reply:
that part moved to blob processor.